.gitlab-ci.yml

```diff
@@ -136,19 +136,11 @@ prepare:front:
   only:
     variables:
       - $THEME_PATH
-  cache:
-    key:
-      files:
-        - ${THEME_PATH}/package.json
-        - ${THEME_PATH}/yarn.lock
-    paths:
-      - ${THEME_PATH}/node_modules/ # Populated during yarn install
```

Comment:
But that's what we should cache! Also needs push/pull setting for cache

Reply:
No it's not. Yarn module install happens not in `prepare:front` but in `sniffers:front`

Reply:
Yes, that's why it should use "pull" here

Reply:
What would it be useful for ? Cache should not be used to pass files from one step to the next. It's the role of [artifacts](https://docs.gitlab.com/ee/ci/caching/#cache-vs-artifacts) and [dependencies](https://docs.gitlab.com/ee/ci/yaml/#dependencies), like used [here](https://github.com/davidferlay/skilld-docker-container/blob/5217d954cd0ddd0c4b9b98b02040e3be1b10c96b/.gitlab-ci.yml#L139)

Reply:
That's what is wrong - cache is supposed to pass files between stages not artifacts!
Artifacts is separate thing - it supposed to be stored in Gitlab itself and should be used to build releases

Reply:
[Quote from documentation](https://docs.gitlab.com/ee/ci/caching/#cache-vs-artifacts) : 
> While the cache could be configured to pass intermediate build results between stages, this should be done with artifacts instead.

Quote about cache : 
> Caches are used to speed up runs of a given job (...) by storing downloaded dependencies so that they don’t have to be fetched from the internet again (like npm packages, Go vendor packages, etc.)

Quote about artifacts : 
> Use for stage results that will be passed between stages. Artifacts are files generated by a job which are stored (...) can then be fetched and used by jobs in later stages of the same pipeline.

Artifacts can indeed also be downloaded by UI

Reply:
Yes, that's why we should cache node modules and put into artefacts only deployable files, otherwise the most expensive (download & unpack) still will happen twice in lint & front